flow/model/qvalue/avro_converter.go

```diff
@@ -509,9 +520,17 @@ func (c *QValueAvroConverter) processNumeric(num decimal.Decimal) any {
 	return rat
 }
 
-func (c *QValueAvroConverter) processBytes(byteData []byte) interface{} {
-	if c.TargetDWH == protos.DBType_CLICKHOUSE {
-		encoded := base64.StdEncoding.EncodeToString(byteData)
+func (c *QValueAvroConverter) processBytes(byteData []byte, format peerdbenv.BinaryFormat) interface{} {
+	if c.TargetDWH == protos.DBType_CLICKHOUSE && format != peerdbenv.BinaryFormatRaw {
+		var encoded string
+		switch format {
+		case peerdbenv.BinaryFormatBase64:
+			encoded = base64.StdEncoding.EncodeToString(byteData)
+		case peerdbenv.BinaryFormatHex:
+			encoded = strings.ToUpper(hex.EncodeToString(byteData))
+		default:
+			panic(fmt.Sprintf("unhandled binary format: %d", format))
```

Comment:
a panic can cause issues with maintenance mode (relies on flow-worker being active to pause mirrors)

Reply:
panic can never happen unless a logic error is introduced in code, & bad code can always panic anyways